service/docs/source/geopm_pio_nvml.7.rst

```diff
@@ -0,0 +1,205 @@
+
+
+      *  ``Aggregation``: Sum
+      *  ``Domain``: GPU
+      *  ``Format``: Double
+      *  ``Unit``: Bytes/Second
+``NVML::GPU_CPU_ACTIVE_AFFINITIZATION``
+    GPU associated with the specified cpu as determined by querying active processes on the GPU.  If no GPUs map to the CPU -1 is returned.  If multiple GPU map to the CPU NAN is returned.
+
+
+      *  ``Aggregation``: Expect_same
+      *  ``Domain``: CPU
+      *  ``Format``: Double
+      *  ``Unit``: None
+``NVML::GPU_MEMORY_UTILIZATION``
+    Fraction of time the GPU memory was accessed in the last set of driver samples
+
+
+      *  ``Aggregation``:
+      *  ``Domain``: GPU
+      *  ``Format``: Double
+      *  ``Unit``: None
+``NVML::GPU_FREQUENCY_MAX_AVAIL``
+    Streaming Multiprocessor maximum frequency in hertz
+
+
+      *  ``Aggregation``: Expect_same
+      *  ``Domain``: GPU
+      *  ``Format``: Double
+      *  ``Unit``: Hertz
+``NVML::GPU_FREQUENCY_MIN_AVAIL``
+    Streaming Multiprocessor minimum frequency in hertz
+
+
+      *  ``Aggregation``: Expect_same
+      *  ``Domain``: GPU
+      *  ``Format``: Double
+      *  ``Unit``: Hertz
+
+Signal Aliases
+~~~~~~~~~~~~~~~~
+Several high level aliases are provided.  Their mapping to
+underlying IO Group signals is provided below.
+
+``GPU_POWER``
+    Aliases to NVML::GPU_POWER
+
+``GPU_CORE_FREQUENCY_STATUS``
+    Aliases to NVML::GPU_FREQUENCY_STATUS
+
+``GPU_CORE_FREQUENCY_MIN_AVAIL``
+    Aliases to NVML::GPU_FREQUENCY_MIN_AVAIL
+
+``GPU_CORE_FREQUENCY_MAX_AVAIL``
+    Aliases to NVML::GPU_FREQUENCY_MAX_AVAIL
+
+``GPU_ENERGY``
+    Aliases to NVML::GPU_ENERGY_CONSUMPTION_TOTAL
+
+``GPU_TEMPERATURE``
+    Aliases to NVML::GPU_TEMPERATURE
+
+``GPU_UTILIZATION``
+    Aliases to NVML::GPU_UTILIZATION
+
+Controls
+--------
+Every control is exposed as a signal with the same name.  The relevant signal aggregation information is provided below.
+
+``NVML::GPU_FREQUENCY_CONTROL``
+    Sets Streaming Multiprocessor frequency min and max to the same limit (in hertz)
+
+
+      *  ``Aggregation``: Average
+      *  ``Domain``: GPU
+      *  ``Format``: Double
+      *  ``Unit``: Hertz
+``NVML::GPU_FREQUENCY_RESET_CONTROL``
+
+    Resets Streaming Multiprocessor frequency min and max limits to default values.  Parameter provided is unused.
+
+      *  ``Aggregation``: Average
+      *  ``Domain``: GPU
+      *  ``Format``: Double
+      *  ``Unit``: None
+``NVML::GPU_POWER_LIMIT_CONTROL``
+    Sets GPU power limit in watts
+
+
+      *  ``Aggregation``: Double
+      *  ``Domain``: GPU
+      *  ``Format``: Sum
+      *  ``Unit``: Watts
+
+Control Aliases
+~~~~~~~~~~~~~~~~
+Several high level aliases are provided.  Their mapping to
+underlying IO Group signals is provided below.
+
+* ``GPU_POWER_LIMIT_CONTROL``
+    Aliases to NVML::GPU_POWER_LIMIT_CONTROL
+
+* ``GPU_CORE_FREQUENCY_CONTROL``
+    Aliases to NVML::GPU_FREQUENCY_CONTROL
+
+
+See Also
```

Comment:
Include a pointer to NVML documentation.  Many of the descriptions above allude to this.